lib/elixir/lib/kernel.ex

```diff
@@ -3627,7 +3627,9 @@ defmodule Kernel do
           unquote(expanded).__using__(unquote(opts))
         end
       _otherwise ->
-        raise ArgumentError, "invalid arguments for use, expected an atom or alias as argument"
+        IO.inspect module
+        IO.inspect __CALLER__
+        raise ArgumentError, "invalid arguments for use, expected a compile time atom or alias, got: #{Macro.to_string(module)}"
```

Comment:
It's only in master and will be available in v1.2. There is no need to dig any deeper, `_otherwise` contains that wrong part of argument.